File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var sortByBits = function(arr) {
  const hamming = n => !n ? 0 : 1+hamming(n&(n-1))
  arr.sort((a, b) => (hamming(a)*100000+a - hamming(b)*100000-b))
  return arr
};

File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.go
func bits(num int) int {
  if num == 0 {
    return 0
  }
  return 1 + bits(num & (num-1))
}

func sortByBits(arr []int) []int {
  sort.Slice(arr, func(i, j int) bool {
    return bits(arr[i]) << 16 | arr[i] < bits(arr[j]) << 16 | arr[j]
  })    
  return arr
}